service/controller/chart/resource/release/update.go

```diff
@@ -254,28 +254,17 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 		}
 	}
 
-	// The release is failed and the values and version have not changed. So we
-	// don't update. We will be alerted so we can investigate.
-	if isReleaseFailed(currentReleaseState, desiredReleaseState) {
-		r.logger.Debugf(ctx, "the %#q release is in status %#q and cannot be updated", desiredReleaseState.Name, currentReleaseState.Status)
+	// We check the controller context and if the release has failed more
+	// than the max number of attempts we stop updating. Otherwise too many
+	// release secrets will be created.
+	//
+	// The releasemaxhistory resource handles deleting the oldest secret if
+	// it is over 1 minute old. So we still retry but at a slower rate.
+	if currentReleaseState.Status == helmclient.StatusFailed && cc.Status.Release.FailedMaxAttempts {
+		r.logger.Debugf(ctx, "the %#q release is in status %#q and has failed %d times", desiredReleaseState.Name, currentReleaseState.Status, project.ReleaseFailedMaxAttempts)
```

Comment:
We should retry even if values or version are unchanged for situations like failed webhooks.

This should mean we get less pages but we also slow the retries and still ensure that too many release secrets are not created.